File: Application/SkuciSe/app/src/main/java/com/example/skucise/models/Advert.kt
package com.example.skucise.models

import com.example.skucise.models.HelperModels.AdvertType
import java.io.Serializable

class Advert(
	var id: Long?,
    var name: String,
    var price: Long,
    var ratingsCount: Int = 0,
    var advertAverageRating: Double = 0.0,

    // Foreign entities
    var userId: Long?,
    var user: User?,
    var advertDetailsId: Long?,
    var advertDetails: AdvertDetails?,
    var realtyTypeId: Long?,
    var advertTypeId: Long?,
    var advertType: AdvertType?,
    var advertImages: List<AdvertImage>?,
    var comments: List<Comment>
) : Serializable {
    constructor() : this(
        0,
        "",
        0,
        0,
        0.0,
        0,
        null,
        0,
        null,
        0,
        0,
        null,
        null,
        listOf()
    )
}

File: Application/Backend/SkuciSeAPI/SkuciSeAPI/Models/Advert.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace SkuciSeAPI.Models
{
    public class Advert
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public long Price { get; set; }
        public int RatingsCount { get; set; }
        public double AdvertAverageRating { get; set; }

        // Foreign Keys
        public long UserId { get; set; }
        public User User { get; set; }

        public long AdvertDetailsId { get; set; }
        public AdvertDetails AdvertDetails { get; set; }

        public long RealtyTypeId { get; set; }
        public RealtyType RealtyType { get; set; }

        public long AdvertTypeId { get; set; }
        public AdvertType AdvertType { get; set; }

        public List<Like> Likes { get; set; }

        public List<AdvertImage> AdvertImages { get; set; }

        public List<Comment> Comments { get; set; }
    }
}
